specs/phase1/validator.md

```diff
@@ -135,56 +135,82 @@ Up to `MAX_EARLY_DERIVED_SECRET_REVEALS`, [`EarlyDerivedSecretReveal`](./custody
 
 ##### Shard transitions
 
-Exactly `MAX_SHARDS` [`ShardTransition`](./beacon-chain.md#shardtransition) objects are included in the block. Default each to an empty `ShardTransition()`. Then for each committee assigned to the slot with an associated `committee_index` and `shard`, set `shard_transitions[shard] = full_transitions[winning_root]` if the committee had enough weight to form a crosslink this slot.
+Up to `MAX_SHARDS` [`ShardTransition`](./beacon-chain.md#shardtransition) objects are included in the block
+(at most one per shard, and must be sorted by shard, numerically increasing).
+A `ShardTransition` inclusion signals sufficient committee support for crosslink to be processed.
+In the event that this block triggers multiple committee thresholds for the same shard to be met,
+the `ShardTransition` for the committee from the latest slot should be included.
 
-Specifically:
-* Call `shards, winning_roots = get_shard_winning_roots(state, block.body.attestations)`
-* Let `full_transitions` be a dictionary mapping from the `shard_transition_root`s found in `attestations` to the corresponding full `ShardTransition`
-* Then for each `shard` and `winning_root` in `zip(shards, winning_roots)` set `shard_transitions[shard] = full_transitions[winning_root]`
+That is, for each shard, search `ShardTransitionCandidate`s in state from most recent to least recent slot
+for sufficient committee participation,
+going no further than those for slots equal to `state.shard_states[shard].slot + 1`.
+
+If sufficient votes have been made for any `ShardTransitionCandidate` and the validator's node knows the 
+`ShardTransition` associated with `shard_transition_candidate.shard_transition_root`, stop the search for that shard
+and add the `shard_transition` to `block.shard_transitions`.
 
-*Note*: The `state` passed into `get_shard_winning_roots` must be transitioned the slot of `block.slot` to run accurately due to the internal use of `get_online_validator_indices` and `is_on_time_attestation`.
+
+Specifically:
+* Let `shard_transitions = []`
+* For each `shard` in `range(get_active_shard_count(state))`
+    * Call `winning_roots = get_shard_winning_roots(state, shard)`
```

Comment:
What's the reason for returning multiple winning roots if `select_transition_from_winning_roots` were only going to choose one. (e.g. the one with lowest index and highest slot )

Reply:
Ah. I see it. The winning root may not be in `full_transitions`.

```python
 if root not in full_transitions:
            continue
```